src/ipfsd-in-proc.js

```diff
@@ -63,8 +59,14 @@ class InProc extends EventEmitter {
         throw new Error(`Unknown argument ${arg}`)
       }
     })
+  }
 
-    this.exec = new IPFS({
+  setExec (cb) {
+    if (this.api !== null) {
+      return setImmediate(() => cb(null, this))
+    }
+    const IPFS = this.opts.exec
+    this.api = this.exec = new IPFS({
```

Comment:
why do we need both if we equal both when running `setExec`, which seems the only place where `this.exec` value is modified.